### ISSUE TITLE
Northern Beaches Council: No data received since 16 Apr 2020

### ISSUE THREAD

**Issue by mlandauer** [planningalerts-scrapers/issues#97]
This issue has been **automatically** created by PlanningAlerts for [Northern Beaches Council](http://www.planningalerts.org.au/authorities/northern_beaches)

It uses the scraper [planningalerts-scrapers/multiple_icon](https://morph.io/planningalerts-scrapers/multiple_icon)

Only close this issue once the authority is working again on PlanningAlerts. Otherwise a new issue will just automatically created. Also, if there is a duplicate issue close the other one in favour of this one.

<img src="https://frontapp.com/assets/img/favicons/favicon-32x32.png" height="16" width="16" alt="Front logo" /> [Front conversations](https://app.frontapp.com/open/top_39mh1)

**Comment by mlandauer** [comment]
Northern beaches has a large population. So fixing this should be a priority. I wonder if we should try contacting the council directly about the xml problem?

**Comment by mlandauer** [comment]
The problem (taken from #65):

This URL works

https://eservices.northernbeaches.nsw.gov.au/ePlanning/live/Public/XC.Track/SearchApplication.aspx?d=thismonth&k=LodgementDate&t=DevApp

but when you stick "&o=xml" on the end (which works with other similar systems at other councils):

https://eservices.northernbeaches.nsw.gov.au/ePlanning/live/Public/XC.Track/SearchApplication.aspx?d=thismonth&k=LodgementDate&t=DevApp&o=xml

**Comment by mlandauer** [comment]
@katska any chance you could pick up contacting the council?

**Comment by katska** [comment]
@mlandauer What is it the council needs to do to unblock us - are they explicitly blocking us, or is it that they're doing something that is a general block of some kind that is affecting us?

**Comment by mlandauer** [comment]
@katska They're not explicitly blocking us as far as I can tell. Their icon software is supposed to support this "o=xml" on the end of the url for doing a search which should return the same data in xml format but when you do that it causes an error (see above). So, probably it's some kind of misconfiguration of their software that they probably don't even realise because the xml feature is rarely used (but **we** use it).

**Comment by katska** [comment]
I could get in touch with them about this!

**Comment by mlandauer** [comment]
@katska yes please get in touch with them. That would be super helpful!

**Comment by kat** [comment]
I've just now written to them to see if it's fixable from council side.

—
Katherine Szuminska

--- original message ---
On August 16, 2021, 3:23 PM GMT+10 ***@***.*** wrote:



@katska yes please get in touch with them. That would be super helpful!

—

You are receiving this because you are subscribed to this thread.

Reply to this email directly, view it on GitHub, or unsubscribe.

Triage notifications on the go with GitHub Mobile for iOS or Android.
--- end of original message ---

**Comment by MutazAshhab** [comment]
@kat Have you heard back from the council regarding this issue?

**Comment by katska** [comment]
@Taz17 Thanks for the ping! I haven't heard back from them except an autoreply saying they'd respond within 10 days (about a month ago). That said it's been the summer hols, so I just emailed them again to see if anyone is around to answer.

**Comment by katska** [comment]
This just in "We have reviewed your request and consequently made some amendments to facilitate supporting the XML output again. You should now find that the service is working." 

The Good News is that the scraper produced this output today, finding 13 DAs. However, the less good news is that of those 13 only 5 show up on PlanningAlerts (see[ recent DAs for Northern Beaches](https://www.planningalerts.org.au/authorities/northern_beaches/applications)) I entered the street address in a fresh search in planningalerts, in case it was an existing DA we already showed, but that's not the case. The other 8 don't appear. @Taz17 or @mlandauer do you have any ideas what might be going on?

[From morph.io](https://morph.io/planningalerts-scrapers/multiple_icon)
Collecting feed data for northern_beaches...
Storing DA2022/0098 - 58 Romford Road FRENCHS FOREST NSW 2086 
Storing DA2022/0081 - 94 Toronto Avenue CROMER NSW 2099 
Storing DA2022/0064 - 3 Henry Street DEE WHY NSW 2099 
Storing DA2022/0063 - 17  Lentara Road BAYVIEW NSW 2104 
Storing DA2022/0060 - 47 Gordon Street MANLY VALE NSW 2093 
Storing DA2021/2641 - 153  Balgowlah Road BALGOWLAH NSW 2093 
Storing DA2021/2607 - 2 Surf Road NORTH CURL CURL NSW 2099 
Storing DA2021/2228 - 323 McCarrs Creek Road TERREY HILLS NSW 2084 
Storing DA2022/0034 - 2 Marcus Place FRENCHS FOREST NSW 2086 
Storing DA2021/2626 - 36  Gurney Crescent SEAFORTH NSW 2092 
Storing DA2021/2146 - 35  Kangaroo Street MANLY NSW 2095 
Storing DA2021/2439 - 10 A Upper Gilbert Street MANLY NSW 2095 
Storing DA2021/2601 - 13  Ocean Avenue NEWPORT NSW 2106

**Comment by MutazAshhab** [comment]
Thats good news, however, the issue of 8 of the 13 DA's not appearing sounds like a morph issue.

I ran it locally and there are 15 rows in the data.sqlite file. (the extra two might have been uploaded after the scraper ran)

I also downloaded the data.sqlite file and select all DA's that were scraped in the past day that belong in the `northern_beaches` authority and I got back 13 rows.

**Comment by MutazAshhab** [comment]
@katska I think this issue belongs in the morph repository. The scraper is working fine

**Comment by katska** [comment]
@Taz17 Hmm could you review again please? The scraper is picking up 13, but then planningalerts is only displaying 5. @mlandauer suggests maybe the workaround previously implemented could be affecting the results.

**Comment by MutazAshhab** [comment]
@katska Here i meant I downloaded the data.sqlite file from [morph](https://morph.io/planningalerts-scrapers/multiple_icon) and did not only rely on my local data.sqlite file.
> I also downloaded the data.sqlite file and select all DA's that were scraped in the past day that belong in the `northern_beaches` authority and I got back 13 rows.

With regards to the recent fix on that scraper. It goes unexecuted. The scraper only executes the workaround only if scraping the `xml` fails. After the council updated it, it no longer does.

Here is the query I ran if you'd like to reproduce what I got.

`select count(*) from data where date_scraped == date('now', '-1 day') AND authority_label == 'northern_beaches';`
and 
`select council_reference, address from data where date_scraped == date('now', '-1 day') AND authority_label == 'northern_beaches';`

**Comment by MutazAshhab** [comment]
@katska & @mlandauer  This does not seem to be an issue anymore as of 7/02/22. All [DA's being scraped](https://morph.io/planningalerts-scrapers/multiple_icon) are showing up on [recent applications section on planning alerts](https://www.planningalerts.org.au/authorities/northern_beaches/applications).

Might have been a 1 time thing. I think we can now close this issue.

**Comment by mlandauer** [comment]
Strange. Since on https://www.planningalerts.org.au/authorities it's not showing as being broken anymore as expected this issue should be automatically tagged with "probably fixed". I'm not sure why it isn't. Probably best to wait until it is in case there's another problem lurking somewhere.

**Comment by MutazAshhab** [comment]
@mlandauer It is tagged with `probably fixed` by your account 5 days ago (which seems to be done automatically)

The tagging is shown under this comment https://github.com/planningalerts-scrapers/issues/issues/97#issuecomment-1020800359

**Comment by mlandauer** [comment]
Weird! I must have been looking at an old version or something. Anyway, it's good so I'll close it.
